monai/metrics/active_learning_metrics.py

```diff
@@ -0,0 +1,119 @@
+
+    Args:
+        include_background: Whether to include the background of the spatial image or channel 0 of the 1-D vector
+        spatial_map: Boolean, if set to True, spatial map of variance will be returned corresponding to i/p image dimensions
+        scalar_reduction: reduction type of the metric, either 'sum' or 'mean' can be used
+        threshold: To avoid NaN's a threshold is used to replace zero's
+
+    """
+
+    def __init__(
+        self,
+        include_background: bool = True,
+        spatial_map: bool = False,
+        scalar_reduction: str = "sum",
+        threshold: float = 0.0005,
+    ) -> None:
+        super().__init__()
+        self.include_background = include_background
+        self.spatial_map = spatial_map
+        self.scalar_reduction = scalar_reduction
+        self.threshold = threshold
+
+    def _compute(self, y_pred: torch.Tensor):
+        """
+        Args:
+            y_pred: Predicted segmentation, typically segmentation model output.
+                It must be N-repeats, repeat-first tensor [N,C,H,W,D].
+
+        Returns:
+            Pytorch tensor of scalar value of variance as uncertainty or a spatial map of uncertainty
+
+        """
+        return compute_variance(
+            y_pred=y_pred,
+            include_background=self.include_background,
+            spatial_map=self.spatial_map,
+            scalar_reduction=self.scalar_reduction,
+            threshold=self.threshold,
+        )
+
+
+def compute_variance(
+    y_pred: torch.Tensor, include_background=True, spatial_map=False, scalar_reduction="mean", threshold=0.0005
+):
+    """
+    Args:
+        y_pred: [N, C, H, W, D] or [N, C, H, W] or [N, C, H] where N is repeats, C is channels and H, W, D stand for
+        Height, Width & Depth
+        include_background: Whether to include the background of the spatial image or channel 0 of the 1-D vector
+        spatial_map: Boolean, if set to True, spatial map of variance will be returned corresponding to i/p image dimensions
+        scalar_reduction: reduction type of the metric, either 'sum' or 'mean' can be used
+        threshold: To avoid NaN's a threshold is used to replace zero's
+    Returns:
+        A single scalar uncertainty/variance value or the spatial map of uncertainty/variance
+    """
+
+    # The background utils is only applicable here because instead of Batch-dimension we have repeats here
+    y_pred = y_pred.float()
+
+    if not include_background:
+        y = y_pred
+        # TODO If this utils is made to be optional for 'y' it would be nice
+        y_pred, y = ignore_background(y_pred=y_pred, y=y)
+
+    # Set any values below 0 to threshold
+    y_pred[y_pred <= 0] = threshold
+
+    n_len = len(y_pred.shape)
+
+    if n_len < 4 and spatial_map is True:
+        print(f"Spatial Map set to True for a 1-D array, the provided input is {y_pred.shape}")
+        print("Spatial map requires a 2D/3D image with N-repeats and C-channels")
```

Comment:
```suggestion
        warnings.warn(f"Spatial Map set to True for a 1-D array, the provided input is {y_pred.shape}",
                       "Spatial map requires a 2D/3D image with N-repeats and C-channels")
```

if you add this change, `import warnings` is needed in this file